sources/packages/web/src/components/common/applicationTracker/ApplicationInProgressChangeRequest.vue

```diff
@@ -0,0 +1,216 @@
+import router from "@/router";
+import { ApplicationService } from "@/services/ApplicationService";
+import ConfirmModal from "@/components/common/modals/ConfirmModal.vue";
+
+/**
+ * Possible waiting processes to be displayed to the student.
+ */
+enum WaitingTypes {
+  MinistryApproval = "MinistryApproval",
+  ParentsDeclaration = "ParentsDeclaration",
+  PartnerDeclaration = "PartnerDeclaration",
+  StudentIncomeVerification = "StudentIncomeVerification",
+  ParentsIncomeVerification = "ParentsIncomeVerification",
+  PartnerIncomeVerification = "PartnerIncomeVerification",
+}
+
+export default defineComponent({
+  emits: {
+    changeRequestCancelled: () => true,
+  },
+  components: {
+    ApplicationStatusTrackerBanner,
+    ConfirmModal,
+  },
+  props: {
+    changeRequest: {
+      type: Object as PropType<ChangeRequestInProgressAPIOutDTO>,
+      required: false,
+    },
+    areApplicationActionsAllowed: {
+      type: Boolean,
+      required: false,
+      default: false,
+    },
+  },
+  setup(props, { emit }) {
+    const cancelChangeRequestModal = ref({} as ModalDialog<boolean>);
+    const snackBar = useSnackBar();
+    const cancelChangeRequestLoading = ref(false);
+
+    const viewChangeRequest = async () => {
+      if (!props.changeRequest) {
+        return;
+      }
+      const applicationId = props.changeRequest.applicationId;
+      const applicationProgramYear =
+        await ApplicationService.shared.getApplicationWithPY(applicationId);
+      await router.push({
+        name: StudentRoutesConst.DYNAMIC_FINANCIAL_CHANGE_REQUEST_APP_FORM,
+        params: {
+          selectedForm: applicationProgramYear.formName,
+          programYearId: applicationProgramYear.programYearId,
+          id: applicationId,
+        },
+      });
+    };
+
+    const cancelChangeRequest = async () => {
+      if (!props.changeRequest) {
+        return;
+      }
+      const confirm = await cancelChangeRequestModal.value.showModal();
+      if (!confirm) {
+        return;
+      }
+      try {
+        cancelChangeRequestLoading.value = true;
+        await ApplicationService.shared.applicationCancelChangeRequest(
+          props.changeRequest.applicationId,
+        );
+        emit("changeRequestCancelled");
+      } catch {
+        snackBar.error(
+          "There was an error cancelling your change request. Please try again.",
+        );
+      } finally {
+        cancelChangeRequestLoading.value = false;
+      }
+    };
+
+    /**
+     * Creates the list of processes currently waiting that are
+     * meaningful to be displayed to the student.
+     */
+    const waitingList = computed(() => {
+      if (!props.changeRequest) {
+        return [];
+      }
+      const waitingList: WaitingTypes[] = [];
+      const change = props.changeRequest;
+      if (
+        change.applicationEditStatus ===
+        ApplicationEditStatus.ChangePendingApproval
+      ) {
+        waitingList.push(WaitingTypes.MinistryApproval);
```

Comment:
On status `ChangePendingApproval` we can push status `MinistryApproval` and return ?

Reply:
Yes, but then we will be coupling the status in an unnecessary way. For instance, should we avoid checking the parent's conditions if a partner is present or check the income if the supporting data is not present yet?
I do not see a benefit for this situation.